tests/by-util/test_rm.rs

```diff
@@ -1217,3 +1218,117 @@ fn test_progress_no_output_on_error() {
         .stderr_contains("cannot remove")
         .stderr_contains("No such file or directory");
 }
+
+#[test]
+fn test_one_file_system() {
+    let scene = TestScenario::new(util_name!());
+    let at = &scene.fixtures;
+    let a_b = "a/b";
+    let t_y = "other_partition_tmpdir/y";
+
+    at.mkdir_all(a_b);
+    at.mkdir_all(t_y);
+
+    #[cfg(any(target_os = "linux", target_os = "macos"))]
```

Comment:
can you do one function for linux
and another for macos ?
it is too hard to understand correctly